src/components/Countries.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCountries } from '../redux/countries/countriesSlice';
+import Country from './Country';
+import styles from '../styles/Countries.module.css';
+
+const Countries = () => {
+  const dispatch = useDispatch();
+  const countries = useSelector((state) => state.countries);
+
+  const [continent, setContinent] = useState('all');
+
+  useEffect(() => {
+    if (countries.length === 0) {
+      dispatch(fetchCountries());
+    }
+  }, [countries, dispatch]);
+
+  return (
+    <div>
+      <h1 className={styles.title}>Countries</h1>
+      <div className={styles.dropdown}>
+        <select
```

Comment:
- [Optional] Hi @shasherazi, great job filtering countries by their continent using the select options. I think it could have been much better to filter countries by their respective names. This can improve the user's experience since he/she will only need to introduce the name of the country in the search field to retrieve the country from the API very quickly instead of scrolling down after filtering by continent.